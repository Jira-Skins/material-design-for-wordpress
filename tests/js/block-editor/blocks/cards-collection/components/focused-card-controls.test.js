/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import FocusedCardControls from '../../../../../../assets/src/block-editor/blocks/cards-collection/components/focused-card-controls';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <FocusedCardControls { ...props } /> );
};

const baseProps = {
	cardIndex: 0,
	style: 'masonry',
	numberOfCards: 1,
	onMoveLeftOrUp: jest.fn(),
	onMoveRightOrDown: jest.fn(),
	onRemove: jest.fn(),
};

describe( 'FocusedCardControls', () => {
	afterEach( () => {
		jest.clearAllMocks();
	} );

	it( 'matches snapshot', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when style prop is `list`', () => {
		const props = { ...baseProps };
		props.style = 'list';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when number of cards is more than 1 and the card index less than the number of cards', () => {
		const props = { ...baseProps };
		props.cardIndex = 2;
		props.numberOfCards = 4;
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
