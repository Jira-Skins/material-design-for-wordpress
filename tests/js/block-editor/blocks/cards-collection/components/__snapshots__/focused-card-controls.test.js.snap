// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusedCardControls matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="card-container-controls"
      >
        <button
          aria-label="Move left"
          class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-left-up has-icon"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-arrow-left"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 14L7 9.97 13 6v8z"
            />
          </svg>
        </button>
        <button
          aria-label="Move right"
          class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-right-down has-icon"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-arrow-right"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 6l6 4.03L8 14V6z"
            />
          </svg>
        </button>
        <span
          class="card-number-title"
        >
          Card #1
           
        </span>
        <button
          aria-label="Remove card"
          class="components-button mtb-card-buttons mtb-card-close-button has-icon"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-no"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-container-controls"
    >
      <button
        aria-label="Move left"
        class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-left-up has-icon"
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-arrow-left"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 14L7 9.97 13 6v8z"
          />
        </svg>
      </button>
      <button
        aria-label="Move right"
        class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-right-down has-icon"
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-arrow-right"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 6l6 4.03L8 14V6z"
          />
        </svg>
      </button>
      <span
        class="card-number-title"
      >
        Card #1
         
      </span>
      <button
        aria-label="Remove card"
        class="components-button mtb-card-buttons mtb-card-close-button has-icon"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-no"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FocusedCardControls matches snapshot when number of cards is more than 1 and the card index less than the number of cards 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="card-container-controls"
      >
        <button
          aria-label="Move left"
          class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-left-up has-icon"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-arrow-left"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 14L7 9.97 13 6v8z"
            />
          </svg>
        </button>
        <button
          aria-label="Move right"
          class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-right-down has-icon"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-arrow-right"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 6l6 4.03L8 14V6z"
            />
          </svg>
        </button>
        <span
          class="card-number-title"
        >
          Card #3
           
        </span>
        <button
          aria-label="Remove card"
          class="components-button mtb-card-buttons mtb-card-close-button has-icon"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-no"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-container-controls"
    >
      <button
        aria-label="Move left"
        class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-left-up has-icon"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-arrow-left"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 14L7 9.97 13 6v8z"
          />
        </svg>
      </button>
      <button
        aria-label="Move right"
        class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-right-down has-icon"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-arrow-right"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 6l6 4.03L8 14V6z"
          />
        </svg>
      </button>
      <span
        class="card-number-title"
      >
        Card #3
         
      </span>
      <button
        aria-label="Remove card"
        class="components-button mtb-card-buttons mtb-card-close-button has-icon"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-no"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FocusedCardControls matches snapshot when style prop is \`list\` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="card-container-controls"
      >
        <button
          aria-label="Move up"
          class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-left-up has-icon"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-arrow-left"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 14L7 9.97 13 6v8z"
            />
          </svg>
        </button>
        <button
          aria-label="Move down"
          class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-right-down has-icon"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-arrow-right"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 6l6 4.03L8 14V6z"
            />
          </svg>
        </button>
        <span
          class="card-number-title"
        >
          Card #1
           
        </span>
        <button
          aria-label="Remove card"
          class="components-button mtb-card-buttons mtb-card-close-button has-icon"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-no"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-container-controls"
    >
      <button
        aria-label="Move up"
        class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-left-up has-icon"
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-arrow-left"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 14L7 9.97 13 6v8z"
          />
        </svg>
      </button>
      <button
        aria-label="Move down"
        class="components-button mtb-card-buttons mtb-card-move-button mtb-card-move-button-right-down has-icon"
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-arrow-right"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 6l6 4.03L8 14V6z"
          />
        </svg>
      </button>
      <span
        class="card-number-title"
      >
        Card #1
         
      </span>
      <button
        aria-label="Remove card"
        class="components-button mtb-card-buttons mtb-card-close-button has-icon"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-no"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
